stock-tracker/src/features/keyStats/redux/keyStatsReducer.test.ts

```diff
@@ -0,0 +1,60 @@
+import { keyStatsReducer, KeyStatsState } from "./keyStatsReducer";
+import { SET_RESPONSE, SET_ERROR_KEYSTATS } from "./actionTypes";
+import { ADD_SYMBOL } from "../../search/redux/actionTypes";
+import { SetSymbol } from "../../search/redux/actions";
+import { Response, KeyStatsActions } from "./actions";
+
+describe("testing keyStats reducer", () => {
+  let initialState: KeyStatsState = {
+    response: undefined,
+    loading: false,
+    error: false
+  };
+
+  let newState: KeyStatsState;
+
+  describe("returns loading as true after the ADD_SYMBOL action is called", () => {
+    const action: SetSymbol = {
+      type: ADD_SYMBOL,
+      payload: "AAPL"
+    };
+
+    beforeAll(() => {
+      newState = keyStatsReducer(initialState, action);
+    });
+
+    it("should update the state", () => {
+      expect(newState.loading).toBeTruthy();
```

Comment:
nit: think it's better to be more explicit on the loading test by using `expect(newState.loading).toEqual(true);` as truthy does not only mean true.

**The following values are always falsy:**

- false
- 0 (zero)
- '' or "" (empty string)
- null
- undefined
- NaN

**Everything else is truthy. That includes:**

-'0' (a string containing a single zero)
- 'false' (a string containing the text “false”)
- [] (an empty array)
- {} (an empty object)
- function(){} (an “empty” function)

[](https://www.sitepoint.com/javascript-truthy-falsy/)

Reply:
Ok thanks, I didn't know that!